cmake/SetupPackages.cmake

```diff
@@ -91,6 +91,11 @@ if (RAJA_ENABLE_TESTS)
       googletest
       GIT_REPOSITORY https://github.com/google/googletest.git
       GIT_TAG release-1.7.0
+      CMAKE_ARGS                
+          -DCMAKE_C_COMPILER=${CMAKE_C_COMPILER}
```

Comment:
It should pull whatever the value determined for the cache is, since it's evaluated at runtime (or it's supposed to be) but it's worth testing it.